### ISSUE TITLE
Choose CSS Framework

### ISSUE THREAD

**Issue by flashback2k14** [socobo/SocoboPlusPlus#65]
- MDL
- Bootstrap 4
- Angular Material 2
- Flexbox Lib only --> write own styles 
- None

@hetec

**Comment by hetec** [comment]
I think we should definitivly use some kind of grid System because it could be really hard to solve all the layouting Problems only with flexbox

**Comment by hetec** [comment]
In the other hand i would Like to habe our own set of css components. Maybe with our completly own styles or a custom bootstrap CSS File or the like.

I would like this because  I would like to have a completly individual style of our app. But this ist a Lot harder than using the mature components of material design or so

**Comment by hetec** [comment]
Whats your opinion?

**Comment by hetec** [comment]
Another question is if we want to use sass and jade or some other preprocessor?

**Comment by flashback2k14** [comment]
I'm with you that we should use an grid layout framework. I'm not sure if its a good idea to write all styles by our self. There is so much things we need to know. I'm not a css pro. I don't know if we need a preprocessor.

**Comment by hetec** [comment]
I would like to use at least sass (or some other css preprocessor) to better organize our css. I think things like a much better syntax and variables (very important in my opinion to organize sizes, colors and so on) are very convenient. I don't really need something like jade or so since angular has already a lot of templating stuff

**Comment by hetec** [comment]
What a framework would you prefer?

**Comment by flashback2k14** [comment]
i have no glue how to work with sass. we should try and see where it goes. 

what framework did you mean? a full or a grid framework?

**Comment by flashback2k14** [comment]
i have some grid frameworks for you @hetec :-) What did you think?

- http://bulma.io/
- http://progressivered.com/fla/?d=0&v=0&h=1&s=3&i=000&a=000
- http://flexboxgrid.com/
- http://www.flexboxpatterns.com/home
- http://www.useful.ly/
- http://thednp.github.io/bootstrap.native

**Comment by hetec** [comment]
It doesn't matter what kind of framework, I just wanted to know if you already have a framework in mind which you would like to use?!

All these frameworks would possibly work, but I think we should stick with some well known like bootstrap or Angular Material. But I'm not sure.

**Comment by flashback2k14** [comment]
Yes, but I think you want a different site styling. If not, i prefer angular material.

**Comment by hetec** [comment]
Ok, what if we use angular material as the basis for the app design, but we don't follow exactly the material Design guide lines? So we could use the grid and components like forms and buttons and at the same time we would add some custom flavor due to using some more individual headers, menus and so on.

What do you think? Would this be ok for you?

**Comment by flashback2k14** [comment]
I think this sounds good.

**Comment by flashback2k14** [comment]
@hetec to recap: we use angular material with custom flavor in mind!?

**Comment by hetec** [comment]
yes I would say that's still the best choice! This [one](https://material.angular.io/components/form-field/overview) ?

**Comment by flashback2k14** [comment]
Yes, this the official angular material site.
